src/ripple/core/impl/ThreadEntry.cpp

```diff
@@ -22,47 +22,49 @@
 #include <ripple/basics/Log.h>
 
 #include <boost/coroutine/exceptions.hpp>
+#include <boost/thread/tss.hpp>
 #include <exception>
 #include <iostream>
 
 namespace ripple {
 
 #ifndef NO_LOG_UNHANDLED_EXCEPTIONS
-namespace detail {
-thread_local
-std::string threadName;
+static boost::thread_specific_ptr<std::string> threadName;
 
+namespace detail {
 void setThreadName(std::string name)
 {
-    threadName = std::move(name);
+    threadName.reset(new std::string(std::move(name)));
```

Comment:
This is, at best, a super-minor nit: could this leak if `reset` throws and does it ever? I guess we have bigger problems if it does.